homeassistant/components/ecovacs/__init__.py

```diff
@@ -38,32 +37,46 @@
     extra=vol.ALLOW_EXTRA,
 )
 
-ECOVACS_DEVICES = "ecovacs_devices"
-
-# Generate a random device ID on each bootup
-ECOVACS_API_DEVICEID = "".join(
-    random.choice(string.ascii_uppercase + string.digits) for _ in range(8)
-)
+PLATFORMS = [
+    Platform.VACUUM,
+]
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Ecovacs component."""
-    _LOGGER.debug("Creating new Ecovacs component")
+    if DOMAIN in config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+            )
+        )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
 
     def get_devices() -> list[VacBot]:
-        ecovacs_api = EcoVacsAPI(
-            ECOVACS_API_DEVICEID,
-            config[DOMAIN].get(CONF_USERNAME),
-            EcoVacsAPI.md5(config[DOMAIN].get(CONF_PASSWORD)),
-            config[DOMAIN].get(CONF_COUNTRY),
-            config[DOMAIN].get(CONF_CONTINENT),
-        )
-        ecovacs_devices = ecovacs_api.devices()
-        _LOGGER.debug("Ecobot devices: %s", ecovacs_devices)
+        try:
+            ecovacs_api = EcoVacsAPI(
+                get_client_device_id(),
+                entry.data[CONF_USERNAME],
+                EcoVacsAPI.md5(entry.data[CONF_PASSWORD]),
+                entry.data[CONF_COUNTRY],
+                entry.data[CONF_CONTINENT],
+            )
+            ecovacs_devices = ecovacs_api.devices()
+        except ValueError as ex:
+            _LOGGER.error("Ecovacs login failed due to wrong username/password")
```

Comment:
The config entry helper will log the exception message including the stack trace.
```suggestion
```